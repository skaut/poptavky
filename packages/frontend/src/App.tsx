/** @jsxImportSource @emotion/react */
import type React from "react"
import { Route, Routes } from "react-router-dom"
import useSWR from "swr"

import { Container } from "./components/Container"
import { Navigation } from "./components/Navigation"
import { config } from "./config"
import type { ProjectListings } from "./interfaces/ProjectListings"
import { IssueDetail } from "./pages/IssueDetail/IssueDetail"
import { IssuesList } from "./pages/IssuesList/IssuesList"
import { ProjectDetail } from "./pages/ProjectDetail/ProjectDetail"
import { ProjectsList } from "./pages/ProjectsList/ProjectsList"

export const App = (): React.JSX.Element => {
  const { data, error } = useSWR<ProjectListings, unknown>(
    config.dataApiUrl,
    async (url: string) => {
      const res = await fetch(url)
      return res.json()
    }
  )
  if (error !== undefined) {
    return <div>Nepodařilo se načíst data webu.</div>
  }
  if (!data) {
    return <div>Načítají se data webu.</div>
  }
  const AppNavigation = (): React.JSX.Element => (
    <Navigation
      items={[
        {
          title: "Poptávky",
          link: "/",
        },
        {
          title: "Projekty",
          link: "/projekty",
        },
      ]}
    />
  )
  return (
    <Routes>
      <Route path="/" element={<Container />}>
        <Route
          path="projekty"
          element={
            <div>
              <AppNavigation />
              <ProjectsList data={data} />
            </div>
          }
        />
        <Route
          path=":owner/:project/:issue"
          element={<IssueDetail data={data} />}
        />
        <Route path=":owner/:project" element={<ProjectDetail data={data} />} />
        <Route
          path="/"
          element={
            <div>
              <AppNavigation />
              <IssuesList data={data} />
            </div>
          }
        />
      </Route>
    </Routes>
  )
}
