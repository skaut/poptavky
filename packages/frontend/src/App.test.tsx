import renderer from "react-test-renderer"
import { App } from "./App"

describe("App", () => {
  it("should renders correctly", () => {
    const tree = renderer
      .create(<App />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  /*
  it("should renders correctly with title", () => {
    const tree = renderer
      .create(
        <Button href="http://www.somewhere.com" title="myTitle">
          Link somewhere
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should renders correctly with custom class", () => {
    const tree = renderer
      .create(
        <Button href="http://www.somewhere.com" className="myClassName">
          Link somewhere
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should renders correctly with target self", () => {
    const tree = renderer
      .create(
        <Button href="http://www.somewhere.com" targetSelf>
          Link somewhere
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  */
})
