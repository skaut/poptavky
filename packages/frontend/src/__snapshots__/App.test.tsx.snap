// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders correctly 1`] = `
<div>
  Načítají se data webu.
</div>
`;
